mlrun/api/db/sqldb/db.py

```diff
@@ -600,6 +688,30 @@ def del_artifact(self, session, key, tag="", project=""):
 
         self._delete(session, Artifact, **kw)
 
+    def _delete_artifacts_tags(
+        self,
+        session,
+        project: str,
+        artifacts: typing.List[Artifact],
+        tags: typing.List[str] = None,
+        commit: bool = True,
+    ):
+        artifacts_keys = [str(artifact.key) for artifact in artifacts]
+        query = (
+            session.query(Artifact.Tag)
+            .join(Artifact)
+            .filter(
+                Artifact.project == project,
+                Artifact.key.in_(artifacts_keys),
+            )
+        )
+        if tags:
+            query = query.filter(Artifact.Tag.name.in_(tags))
+        for tag in query:
+            session.delete(tag)
+        if commit:
```

Comment:
No biggie, but in general I would commit only if the query returned anything and there's really something to commit.